README.md

```diff
@@ -150,91 +108,57 @@ Next, add the `LockActivity` inside the `application` tag:
   android:theme="@style/Lock.Theme"/>
 ```
 
-> In versions 2.5.0 or lower of Lock.Android you had to define an **intent-filter** inside the `LockActivity` to make it possible for the library to capture a social provider's authentication result. This intent-filter declaration is no longer required for versions greater than 2.5.0, as it's now done internally by the library for you.
-
-In case you are using an older version of Lock for **Social** Authentication, the **intent-filter** must be added to the `LockActivity` by you.
-
-```xml
-<activity
-  android:name="com.auth0.android.lock.LockActivity"
-  android:label="@string/app_name"
-  android:launchMode="singleTask"
-  android:screenOrientation="portrait"
-  android:theme="@style/Lock.Theme">
-    <intent-filter>
-      <action android:name="android.intent.action.VIEW" />
-
-      <category android:name="android.intent.category.DEFAULT" />
-      <category android:name="android.intent.category.BROWSABLE" />
-
-      <data
-        android:host="@string/com_auth0_domain"
-        android:pathPrefix="/android/${applicationId}/callback"
-        android:scheme="https" />
-    </intent-filter>
-</activity>
-```
+Make sure the Activity's `launchMode` is declared as `singleTask` or the authentication result won't come back into your application.
 
 
-Make sure the Activity's `launchMode` is declared as `singleTask` or the result won't come back in the authentication.
+Then, in any of your Activities, you need to initialize **Lock** and handle the release of its resources appropriately after you're doing using it. 
 
-
-Then, in any of your Activities, you need to initialize **Lock**. If your `auth0Scheme` value is not `https` you need to use `withScheme()` as shown below.
-
-```java
+```kotlin
 // This activity will show Lock
-public class HomeActivity extends Activity {
+class MyActivity : AppCompatActivity() {
 
-  private Lock lock;
+  private var lock: Lock? = null
 
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    // Your own Activity code
-    Auth0 account = new Auth0("YOUR_AUTH0_CLIENT_ID", "YOUR_AUTH0_DOMAIN");
-    account.setOIDCConformant(true);
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    
+    val account = Auth0(this)
     lock = Lock.newBuilder(account, callback)
-      //Customize Lock
-      //.withScheme("myapp")
-      .build(this);
+      // Customize Lock
+      // .withScheme("myapp")
+      .build(this)
   }
 
-  @Override
-  protected void onDestroy() {
-    super.onDestroy();
-    // Your own Activity code
-    lock.onDestroy(this);
-    lock = null;
```

Comment:
Why is there no need to set the reference to null in the Kotlin version?

Reply:
Is not because of kotlin. When an android activity gets its onDestroy method called is because is going away for good. There's no need to set to null the references. The garbage collector will recycle the whole activity instance.